File: todoappbackend/models/todo.js
const mongoose = require("mongoose");


// Define the TODO Schema
const todoSchema = new mongoose.Schema({
  todotitle: {
    type: String,
    required: true,
    trim: true,
  },
  description: {
    type: String,
    default: "",
    trim: true,
  },
  isCompleted: {
    type: Boolean,
    default: false
  },
  priority: {
    type: String,
    enum: ["low", "medium", "high"],
    default: "medium",
  },
},
  { timestamps: true },
);

const Todos = mongoose.model('Todo', todoSchema);
module.exports = Todos;

File: todoappflutter/lib/models/todo.dart
class Todo {
  final String id;
  String title;
  String description;
  bool isCompleted;
  final DateTime createdAt;
  String priority; // High, Medium, Low

  Todo({
    required this.id,
    required this.title,
    required this.description,
    this.isCompleted = false,
    required this.createdAt,
    required this.priority
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'todotitle': title,
      'description': description,
      'isCompleted': isCompleted,
      'createdAt': createdAt.toIso8601String(),
      'priority': priority,
    };
  }

  factory Todo.fromJson(Map<String, dynamic> json) {
    return Todo(
      id: json['_id'] ?? '',
      title: json['todotitle'] ?? '',
      description: json['description'] ?? '',
      isCompleted: json['isCompleted'] ?? false,
      createdAt: DateTime.parse(json['createdAt']),
      priority: json['priority'] ?? 'medium',
    );
  }
}
